external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -78,7 +86,7 @@ struct Context<'env> {
     local_scopes: Vec<BTreeMap<Symbol, u16>>,
     local_count: BTreeMap<Symbol, u16>,
     used_locals: BTreeSet<N::Var_>,
-    nominal_blocks: Vec<(Symbol, u16, NominalBlockType)>,
+    nominal_blocks: Vec<(Loc, Option<Symbol>, u16, NominalBlockType)>,
```

Comment:
Given all of the code shifting, I wonder if it would be better to create the labels up front. In particular this revision and the one below:

```suggestion
    nominal_blocks: Vec<(Option<Symbol>, BlockLabel, NominalBlockType)>,
```